### PR TITLE
blockdev_commit_fio: fix background thread can't exit preperly

### PR DESCRIPTION
**Pull request by yanan-fu**
Need ensure all backgroupd threads exit before the main thread exits,
otherwise, will hit error "Test reported status but did not finish"

id: 2072801
Signed-off-by: Yanan Fu <yfu@redhat.com>

### CODE REVIEW

**Comment by zhencliu** [comment]
LGTM

File: qemu/tests/blockdev_commit_fio.py [hunk 1]
```diff
@@ -50,7 +46,9 @@ def run_test(self):
             self.verify_data_file()
         finally:
             if bg_test.is_alive():
+                # clean fio cause fio.run exit, then fio thread exit
```
[the remark this answers is not in the document]
Comment (yanan-fu) [reply qemu/tests/blockdev_commit_fio.py:49]: Make sense, let me update it, thanks!

**Comment by yanan-fu** [comment]
Repeat 50 times with `blockdev_snapshot_reoot`, all pass.

But test with `blockdev_commit_fio` failed sometimes. From my checking, `killall fio` can not kill all the 256 fio process as we use  `--numjobs=256`.

**Comment by yanan-fu** [comment]
Repeat test pass in my side.  @zhencliu @aliang123 Could you help review ? Thanks!

**Comment by zhencliu** [comment]
LGTM

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_commit_fio.q35: PASS (355.07 s)

Acked-by: Aihua Liang <aliang@redhat.com>

**Comment by yanan-fu** [comment]
Thanks all!

**Pull request merged**